.github/workflows/test.yaml

```diff
@@ -19,12 +19,10 @@ jobs:
             
       - run: flutter --version
       - run: flutter pub get
-
-      - name: Verify formatting
-        run: dart format --output=none .
+      - run: flutter pub get --directory=example/viam_example_app
 
       - name: Analyze project source
-        run: dart analyze --fatal-infos
+        run: dart analyze --no-fatal-warnings
```

Comment:
On default, `dart analyze` will fail if it finds `warnings`. `--no-fatal-warnings` causes the check to not fail, this ought to be removed eventually to enforce failures on warnings, but for now we have a few that need to be cleaned up.

I would even further prefer adding  `--fatal-infos` which enforces failures on the `info` level, it will take a bit more work to get there as well. In a few iterations I think we can be there.

Reply:
SGTM; could optionally add a `TODO` here with ticket num for a ticket that removes all warnings and infos.